dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtilsTest.java

```diff
@@ -794,4 +809,163 @@ public void testSortPullRelated(final SortTestCase testCase) throws DotDataExcep
         }
     }
 
+    /**
+     * Method to test: {@link ContentUtils#pull(String, int, String, User, String)}
+     * When: there is a content with a publish date in the future and the time machine parameter in null
+     * Should: Not return the content
+     */
+    @Test
+    public void whenTheTimeMachineDateIsNullAndPublishDateInFutureShouldNotReturnAnything() {
+        final String timeMachine = null;
+        final Calendar contentPublishDate = Calendar.getInstance();
+        contentPublishDate.add(Calendar.DATE, 1);
+
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+        new ContentletDataGen(contentType.id())
+                .setPolicy(IndexPolicy.FORCE)
+                .setProperty("sysPublishDate", contentPublishDate.getTime())
+                .nextPersisted();
+
+        final String query = String.format(QUERY_BY_STRUCTURE_NAME, contentType.variable());
+
+        final List<Contentlet> contentlets = ContentUtils.pull(query, 10, null, APILocator.systemUser(), timeMachine);
+
+        assertEquals(0, contentlets.size());
+
+        final List<Contentlet> workingContent = ContentUtils.pull(query.replace("live", "working"),
+                10, null, APILocator.systemUser(), null);
+        assertEquals(1, workingContent.size());
+    }
+
+    /**
+     * Method to test: {@link ContentUtils#pull(String, int, String, User, String)}
+     * When: there is a content with a publish date set to tomorrow and the time machine date is the date after tomorrow
+     * Should: return one content
+     */
+    @Test
+    public void whenTheTimeMachineDateAndPublishDateAreTomorrowShouldReturnOneContent() {
+        final Calendar publishDate = Calendar.getInstance();
+        publishDate.add(Calendar.DATE, 1);
+
+        final Calendar timeMachine = Calendar.getInstance();
+        timeMachine.add(Calendar.DATE, 2);
+
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+        Contentlet contentlet = new ContentletDataGen(contentType.id())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentlet' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4740221)